packages/table/test/common/internal/platformUtilsTests.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ * Licensed under the BSD-3 License as modified (the “License”); you may obtain a copy
+ * of the license at https://github.com/palantir/blueprint/blob/master/LICENSE
+ * and https://github.com/palantir/blueprint/blob/master/PATENTS
+ */
+
+import { expect } from "chai";
+
+import * as PlatformUtils from "../../../src/common/internal/platformUtils";
+
+describe("platformUtils", () => {
+    describe("isMac", () => {
+        it("returns false for non-Mac platforms", () => {
+            expect(PlatformUtils.isMac("Win32")).to.be.false;
+            expect(PlatformUtils.isMac("linux")).to.be.false;
+        });
+
+        it("returns true for Mac platforms", () => {
+            expect(PlatformUtils.isMac("Mac")).to.be.true;
+            expect(PlatformUtils.isMac("iPhone")).to.be.true;
+            expect(PlatformUtils.isMac("iPod")).to.be.true;
+            expect(PlatformUtils.isMac("iPad")).to.be.true;
+        });
+    });
+
+    describe("isModKeyPressed", () => {
+        describe("on non-Mac platform", () => {
+            const PLATFORM = "Win32";
+
+            it("returns true if CTRL key pressed", () => {
+                const fakeEvent: any = { metaKey: false, ctrlKey: true };
+                expect(PlatformUtils.isModKeyPressed(fakeEvent, PLATFORM)).to.be.true;
```

Comment:
You can't do `const isMac = sinon.stub(PlatformUtils, "isMac")` and expect the tests to pass, because under the hood, `isModKeyPressed` invokes `isMac` as its own free-floating function, not as a child of some `PlatformUtils` construct.

Thus, a `platformOverride` parameter.

Reply:
that seems wrong... you should be able to stub the function at the module where it's exported (but not at some other module which re-exports the symbol). did you try stubbing it with a specific implementation (specify its return value)?

Reply:
I'd love to be wrong, but it wouldn't work with my normal strategies. I tried stubbing the return value within the `describe`, but that actually stubs the `isMac` function before the `describe("isMac")` block runs, which causes all those tests to fail. When I did the stubbing in a `beforeEach`, it didn't work. Maybe I was just tired and missing something yesterday - I'll try some other stuff right now.

Reply:
Yeah, this test fails when run on a Mac; `isModKeyPressed` still invokes the real `isMac` function in its same file:

```tsx
import * as PlatformUtils from "../../../src/common/internal/platformUtils";

it("returns true if CTRL key pressed", () => {
    const isMac = sinon.stub(PlatformUtils, "isMac").returns(false);
    const fakeEvent: any = { metaKey: false, ctrlKey: true };
    expect(PlatformUtils.isModKeyPressed(fakeEvent)).to.be.true;
    isMac.restore();
});
```

Reply:
How would I test the logic in the real `isMac` without a `platformOverride` param? I'd like to just stub `navigator.platform`, but you can't because it's not a function.